core/src/test/java/org/apache/iceberg/hadoop/HadoopFileIOTest.java

```diff
@@ -143,24 +158,17 @@ public void testDeleteFilesErrorHandling() {
         .hasMessage("Failed to delete 2 files");
   }
 
-  @Test
-  public void testHadoopFileIOKryoSerialization() throws IOException {
-    FileIO testHadoopFileIO = new HadoopFileIO();
+  @ParameterizedTest
+  @MethodSource("serializers")
+  public void testHadoopFileIOSerialization(
+      TestHelpers.RoundTripSerializerFunction<FileIO> roundTripSerializerFunction)
```

Comment:
minor: I would just omit the `Function` part both in the interface name and in the var name